pubsub/jetstream/jetstream.go

```diff
@@ -88,8 +96,26 @@ func (js *jetstreamPubSub) Features() []pubsub.Feature {
 }
 
 func (js *jetstreamPubSub) Publish(req *pubsub.PublishRequest) error {
-	js.l.Debugf("Publishing topic %v with data: %v", req.Topic, req.Data)
-	_, err := js.jsc.Publish(req.Topic, req.Data)
+	var opts []nats.PubOpt
+	var msgId string
+
+	event, err := unmarshallCloudEvent(req)
+	if err != nil {
+		js.l.Debugf("error unmarshalling cloudevent: %v", err)
+	} else {
+		// Use the cloudevent id as the Nats-MsgId for deduplication
+		msgId = event.Context.GetID()
+		opts = append(opts, nats.MsgId(msgId))
+	}
+
+	if msgId == "" {
+		js.l.Warn("empty message ID, Jetstream deduplication will not be possible")
+	}
```

Comment:
I can remove this if you think it is too verbose (considering those who are writing their own cloudevents or custom structure if that's possible).